graphviz.lua

```diff
@@ -35,20 +40,50 @@ local __Graph = {
 			src = src .. ("\t\t%s [label=\"%s\"]\n"):format(node[i].node, node[i].label)
 		end
 
+		local subgraphs = self.__subgraphs
+		for subgraphname , subgraph in pairs(subgraphs) do
+			src = src .. ("\n%s\n"):format( subgraph:source("subgraph" , subgraphname , level + 1) )
+		end
+
 		local edge = self.edges.edge
 		for i = 1, #edge do
 			src = src .. ("\t\t\t%s -> %s\n"):format(edge[i].prev, edge[i].succ)
 		end
 
-		return src .. "}"
+		src = src .. "}"
+
+		--indent
```

Comment:
in this case `delimiter` is fixed by `"\n"` so the argument is not needed.
More to say, this `split` should be like this:

```lua
local function split(str)
	if not str:match("\n$") then
		str = str .. "\n"
	end

	local ret = {}

	for w in str:gmatch("(.-)\n") do
		table.insert(ret, w)
	end

	return ret
end

local lines = split(src)
......
```